### PR TITLE
update the Drone server template for Drone version 0.5

### PR DESCRIPTION
**Pull request by nrvale0**
- New Cattle templates from Drone Server and Agent version 0.5
  - Now has host label affinity questions. 
  - Now front-ended w/  SSL LB.
  - Will need to be adjust to point to non-nrvale0-namespaced containers (back to rancher/)
- Adjusted Cattle templates for Drone Server and Agent 0.4 to call out version.

@will-chan @cloudnautique

### CODE REVIEW

**Comment by cloudnautique** [comment]
@nrvale0 can you pull in master and see if we get a clean CI run?

**Comment by nrvale0** [comment]
@cloudnautique The CI runs are failing for reasons addressed in another PR #262.

**Comment by nrvale0** [comment]
depends #262

File: templates/drone-agents-v05/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+drone-agent-healthcheck:
+  image: nrvale0/drone-config:0.5
+  tty: true
+  labels:
+    io.rancher.container.pull_image: always
+  net: 'container:drone-agent'
+  volumes_from:
+    - drone-agent-data-volume
+  environment:
+    entrypoint: "/giddyup health"
```
Comment (STRML) [inline templates/drone-agents-v05/0/docker-compose.yml]: This is incorrectly nested

File: templates/drone-v05/0/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+A minimal set of configuration looks something like the following. Details will vary according to your environment:
+
+```ShellSesssion
+export DRONE_DEBUG='true'                       # enable debugging mode
+export DRONE_SECRET='<password>'                # the shared secret to allow auto-registration of Drone Agents
+export DRONE_GITHUB=1                           # enable the Drone GitHub integration
+export DRONE_GITHUB_CLIENT='<GitHub app ID>     # GitHub oAuth application client hash
+export DRONE_GITHUB_SECRET='<GitHub app secret> # GitHub oAuth application secret
+export DRONE_DATABASE_DRIVER='mysql'            # enable the MySQL backend for state storage
```
Comment (STRML) [inline templates/drone-v05/0/README.md]: Probably worth noting the default is sqlite3, which requires no database config at all

File: templates/drone-v05/0/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      type: "string"
+      default: ''
+      require: false
+      label: "host label(s) for cluster scheduling node affinity"
+      description: >-
+        Specificy a host label for node scheduling affinity. Example: If you have labelled the node(s) on which you would like
+        Drone Server to live 'ci.role=drone_server' specify 'ci.role=drone_server' here.
+
+    - variable: "debug_mode"
+      type: "string"
```
Comment (STRML) [inline templates/drone-v05/0/rancher-compose.yml]: This can be a `boolean`

File: templates/drone-v05/0/rancher-compose.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      type: "service"
+      label: "Database Service"
+      description: >-
+        Service to link to for database. For instance if using the default Rancher Galera cluster select galera/galera-lb.
+      default: "galera/galera-lb"
+
+    - variable: "host_affinity"
+      type: "string"
+      default: ''
+      require: false
```
Comment (STRML) [inline templates/drone-v05/0/rancher-compose.yml]: should be `required`, not `require`, and no need for blank default

File: templates/drone-agents-v05/0/rancher-compose.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      type: "string"
+      default: ''
+      required: false
+      label: "host label(s) for cluster scheduling node affinity"
+      description: >-
+        Specificy a host label for node scheduling affinity. Example: If you have labelled the node(s) on which you would like
+        Drone Agent to live 'ci.role=drone_agent' specify 'ci.role=drone_agent' here.
+
+    - variable: "debug_mode"
+      type: "string"
```
Comment (STRML) [inline templates/drone-agents-v05/0/rancher-compose.yml]: boolean

**Comment by rawmind0** [comment]
Hi @nrvale0 ,

I apologize, but i rewrites history git repo by mistake. In order to be able to treat PR, it's needed that you fetch the rewritten history and rebase any local changes into the rewritten history.

I'm sorry, thank so much...

**Pull request closed without merging**